Python-packages/covidcast-py/covidcast/covidcast.py

```diff
@@ -244,6 +244,32 @@ def metadata() -> pd.DataFrame:
     return meta_df
 
 
+def _detect_metadata(data: pd.DataFrame,
+                     data_source_col: str = "data_source",
+                     signal_col: str = "signal",
+                     geo_type_col: str = "geo_type") -> Tuple:
+    """Given a DataFrame, return the signal attributes of that DataFrame.
+
+    Errors if any of the columns are heterogeneous.
```

Comment:
probably better to be explicit and say `"Raises ``ValueError`` if..."`

Reply:
Agreed, added.